src/file_search/model.rs

```diff
@@ -0,0 +1,286 @@
+/// You can create multiple stores to organize your documents.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct FileSearchStore {
+    /// Resource name (e.g., "fileSearchStores/my-store-123")
+    pub name: String,
+
+    /// Human-readable display name (max 512 chars)
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub display_name: Option<String>,
+
+    /// Creation timestamp
+    #[serde(with = "time::serde::rfc3339")]
+    pub create_time: OffsetDateTime,
+
+    /// Last update timestamp
+    #[serde(with = "time::serde::rfc3339")]
+    pub update_time: OffsetDateTime,
+
+    /// Number of active documents
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "deserialize_optional_string_to_i64"
+    )]
+    pub active_documents_count: Option<i64>,
+
+    /// Number of pending documents
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "deserialize_optional_string_to_i64"
+    )]
+    pub pending_documents_count: Option<i64>,
+
+    /// Number of failed documents
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "deserialize_optional_string_to_i64"
+    )]
+    pub failed_documents_count: Option<i64>,
+
+    /// Total size in bytes
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "deserialize_optional_string_to_i64"
+    )]
+    pub size_bytes: Option<i64>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CreateFileSearchStoreRequest {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub display_name: Option<String>,
+}
+
+/// A document represents a file within a file search store.
+///
+/// Documents are automatically chunked, embedded, and indexed when uploaded.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Document {
+    /// Resource name (e.g., "fileSearchStores/*/documents/doc-123")
+    pub name: String,
+
+    /// Human-readable display name (max 512 chars)
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub display_name: Option<String>,
+
+    /// Custom metadata (max 20 items)
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub custom_metadata: Option<Vec<CustomMetadata>>,
+
+    /// Last update timestamp
+    #[serde(with = "time::serde::rfc3339")]
+    pub update_time: OffsetDateTime,
+
+    /// Creation timestamp
+    #[serde(with = "time::serde::rfc3339")]
+    pub create_time: OffsetDateTime,
+
+    /// Current state
+    pub state: DocumentState,
+
+    /// Size in bytes
+    #[serde(deserialize_with = "deserialize_string_to_i64")]
+    pub size_bytes: i64,
+
+    /// MIME type
+    pub mime_type: String,
```

Comment:
Maybe it would be better to be typed?

Reply:
https://docs.rs/mime/latest/mime/

I think we should use `mime::Mime` as struct field with `from_str` deserializer. It'll provide typed interface, but in the same time it'll increase serialization code 🤔